assembly/src/assembler/instruction/procedures.rs

```diff
@@ -88,7 +88,7 @@ impl Assembler {
             // `MastForest` contains all the procedures being called; "external procedures" only
             // known by digest are not currently supported.
             let callee_id = mast_forest
-                .get_node_id_by_digest(mast_root)
+                .find_root(mast_root)
                 .unwrap_or_else(|| panic!("MAST root {} not in MAST forest", mast_root));
```

Comment:
I think the logic here (and in the code below) may need to be refactored. Specifically:

- For `call` instructions, if we don't find the procedure root in the current MAST forest, we should create an `External` node for it and then add a `Call` node that references it.
- A `syscall` should always first create an `External` node and then a `Syscall` node which references it. The idea being that the kernel should always be provided to the VM via the host and there is no need to "embed" it into the program MAST. For this, we'll need to figure out how we want to load kernels into the process - so, maybe leave it for the next PR.
- For `exec` instructions, if we don't find a root in the current MAST forest, we should return an error (and probably not panic). In the future, when we introduce `exec.0x...` syntax, it would work exactly the same way as for `call` instructions.

Reply:
Oh right, sorry I forgot to address this the first time around. Should be fixed now

Reply:
> Note: For `syscall`, I use the same logic as `call` (at least for this PR). Is there a reason why we would want to explicitly disallow the kernel to be compiled in the same `MastForest`?

That's fine for now. The reason for not compiling the kernel with the program is that the VM needs to be loaded with the kernel anyway - so, might as well just use it from there.

Reply:
> For exec instructions, if we don't find a root in the current MAST forest, we should return an error (and probably not panic). In the future, when we introduce exec.0x... syntax, it would work exactly the same way as for call instructions.

@bobbinth That syntax already exists actually. In my opinion, the only difference between `call` and `exec` is that `call` has an explicit node that wraps the callee node, i.e. `Call` - in every other regard they should be handled the same here.